jsonnet/obs-operator.jsonnet

```diff
@@ -31,7 +31,9 @@ local operatorObs = obs {
     } + if std.objectHas(cr.spec, 'rule') then cr.spec.rule else {},
 
     stores+:: {
+      local deleteDelay = if std.objectHas(cr.spec, 'compact') && std.objectHas(cr.spec.compact, 'deleteDelay') then cr.spec.compact.deleteDelay else obs.thanos.compact.config.deleteDelay,
       securityContext: if std.objectHas(cr.spec, 'securityContext') then cr.spec.securityContext else obs.thanos.stores.config.securityContext,
+      ignoreDeletionMarksDelay: std.parseInt(std.substr(deleteDelay, 0, std.length(deleteDelay)-1))/2 + std.substr(deleteDelay, std.length(deleteDelay)-1, std.length(deleteDelay)),
```

Comment:
this has some potential weird edge cases. If someone typed `1d` for the deletion delay, then we would try to parse the int `1/2`, which would result in `0d`. This seems slightly dangerous to do in the operator code since the code runs automatically unlike when using plain jsonnet without an operator where users have the chance to review the generated manifests before applying

Reply:
so do you mean that we should expose `ignoreDeletionMarksDelay` for customer to set?

Reply:
yes, I think this would be a good way to start working around this for now. Otherwise we need to move parsing somewhere like golang